src/Core/System/Snippet/Command/Util/CountryAgnosticFileValidator.php

```diff
@@ -0,0 +1,175 @@
+                'SwagLanguagePack/src/Resources/snippet',
+                'SwagLanguagePack/src/Resources/app/administration/src/snippet',
+                ...$options->getIgnoredPaths(),
+            ])
+            ->name([SnippetPatterns::CORE_SNIPPET_FILE_PATTERN, SnippetPatterns::ADMIN_SNIPPET_FILE_PATTERN])
+            ->sortByName(true);
+
+        if ($options->getDir()) {
+            $finder->in($options->getDir());
+        } elseif (empty($options->getExtensionPaths())) {
+            $finder->in('src');
+
+            if ($options->isAll()) {
+                $finder->in('custom/plugins');
+                $finder->in('custom/apps');
+            }
+        } else {
+            $extensionPaths = \array_reduce($options->getExtensionPaths(), static function (array $accumulator, string $extension): array {
+                $extension = trim($extension);
+                $appPath = 'custom/apps/' . $extension;
+                $pluginPath = 'custom/plugins/' . $extension;
+
+                $isApp = is_dir('custom/apps/' . $extension);
+                $isPlugin = is_dir('custom/plugins/' . $extension);
+
+                if (!$isApp && !$isPlugin) {
+                    throw SnippetException::invalidExtension($extension);
+                }
+
+                if ($isApp) {
+                    $accumulator[] = $appPath;
+                }
+
+                if ($isPlugin) {
+                    $accumulator[] = $pluginPath;
+                }
+
+                return $accumulator;
+            }, []);
+
+            $finder->in($extensionPaths);
+        }
+
+        return $finder;
+    }
+
+    public function checkTranslationFiles(ValidatedTranslationFileOptions $options): ValidatedTranslationFileStruct
+    {
+        $finder = $this->getFinder($options);
+        if ($finder->count() < 0) {
+            return new ValidatedTranslationFileStruct();
+        }
+
+        $languageFiles = new TranslationFileCollection([]);
+        $countrySpecificFileCollection = new TranslationFileCollection([]);
+        foreach ($finder as $file) {
+            $filename = $file->getFilename();
+
+            $isCoreTranslationFile = preg_match(
+                SnippetPatterns::CORE_SNIPPET_FILE_PATTERN,
+                $filename,
+                $coreFileData,
+                \PREG_UNMATCHED_AS_NULL
+            );
+
+            $isAdminTranslationFile = preg_match(
+                SnippetPatterns::ADMIN_SNIPPET_FILE_PATTERN,
+                $filename,
+                $adminFileData,
+                \PREG_UNMATCHED_AS_NULL
+            );
+
+            if (!$isAdminTranslationFile && !$isCoreTranslationFile) {
+                continue;
+            }
+
+            $currentFileData = $isAdminTranslationFile ? $adminFileData : $coreFileData;
+
+            $currentDomain = $currentFileData['domain'] ?? 'administration';
+            $formatedLanguageStruct = new TranslationFile(
```

Comment:
I would move this block also to an own private method: `createTranslationFile` or something. To much code for a public method IMO.